src/goto-programs/remove_const_function_pointers.h

```diff
@@ -0,0 +1,96 @@
+/*******************************************************************\
+
+Module: Goto Programs
+
+Author: Thomas Kiley, thomas.kiley@diffblue.com
+
+\*******************************************************************/
+
+#ifndef CPROVER_GOTO_PROGRAMS_REMOVE_CONST_FUNCTION_POINTERS_H
+#define CPROVER_GOTO_PROGRAMS_REMOVE_CONST_FUNCTION_POINTERS_H
+
+#include "goto_model.h"
+#include <util/message.h>
+#include <util/expr.h>
+#include <util/namespace.h>
+
+
+class remove_const_function_pointerst:public messaget
+{
+public:
+  typedef std::set<exprt> functionst;
```

Comment:
Ah no I realise where this will cause problems - in the approx tests as they will pass even if no match. They can however be fixed without ordering by adding the cases that shouldn't be generated to the exclude lines.